### PR TITLE
Check for partially overlapping sticky ends

### PR DESCRIPTION
**Pull request by dgruano**
Related to issue #6 

This PR modifies the assembly verification step to check for partially overlapping sticky ends. It also ads unit tests for the new functionality.

Remaining TODOs:
- Modify frontend to allow user to decide whether to allow or no partially overlapping overhangs.
- Allow assembly of these overhangs

### CODE REVIEW

**Comment by manulera** [comment]
Also this will be the first non-me contribution to this project! 🏅🏅🏅🏅

**Comment by dgruano** [comment]
It seems I cannot find your suggested changes! I would say you can actually commit to my branch as the option "allow edits by maintainers" is checked. Happy to make the changes myself nevertheless!

**Comment by manulera** [comment]
Ok, did not know about that option! I made the change. See if you agree and I will merge!

**Review by manulera — APPROVED** [review]

File: dna_functions.py [hunk 1]
```diff
@@ -17,12 +17,27 @@
...
+        if type_seq1 != type_seq2 or type_seq2 == "blunt":
+            return False
+        elif type_seq2 == "5'":
+            sticky_seq1 = str(reverse_complement(sticky_seq1))
+        elif type_seq2 == "3'":
+            sticky_seq2 = str(reverse_complement(sticky_seq2))
+    
+        ovhg_len = min(len(sticky_seq1), len(sticky_seq2))
+        for i in range(ovhg_len):
+            if sticky_seq1[-(i+1):] == sticky_seq2[:i+1]:
```
[the remark this answers is not in the document]
Comment (dgruano) [reply dna_functions.py]: Yes, that is cleaner, although it should be `range(1, ovhg_len+1)` then. I modified this added a test, and pushed it.

**Pull request merged**